packages/compiler-core/src/transforms/transformExpression.ts

```diff
@@ -269,33 +277,38 @@ export function processExpression(
 
   walkIdentifiers(
     ast,
-    (node, parent, _, isReferenced, isLocal) => {
-      if (isStaticPropertyKey(node, parent!)) {
+    (id, parent, _, isReferenced, isLocal) => {
+      if (isStaticPropertyKey(id, parent!)) {
         return
       }
       // v2 wrapped filter call
-      if (__COMPAT__ && node.name.startsWith('_filter_')) {
+      if (__COMPAT__ && id.name.startsWith('_filter_')) {
         return
       }
 
-      const needPrefix = isReferenced && canPrefix(node)
+      const needPrefix = isReferenced && canPrefix(id)
       if (needPrefix && !isLocal) {
         if (isStaticProperty(parent!) && parent.shorthand) {
           // property shorthand like { foo }, we need to add the key since
           // we rewrite the value
-          ;(node as QualifiedId).prefix = `${node.name}: `
+          ;(id as QualifiedId).prefix = `${id.name}: `
+        }
+
+        if (parent && isNewExpression(parent)) {
+          node.content = rewriteNewExpression(node.content, id, parent)
```

Comment:
I just closed my PR, keep this one.💪🏻💪🏻💪🏻
there are more test cases to consider:
```javascript
new A(x)
new A(new B())
new A(new B(),x)
```

Reply:
When these test cases are embedded, things become more complicated, I can't directly change node.content, and it is difficult to achieve the desired function through string manipulation, because the end position of NewExpression in the string is uncertain, I should Need more time to solve this, maybe you can give me some advice or help

Reply:
To be honest, I haven't found a good way.😄
I'm not sure it's worth it. in most scenarios, the user won't return a ref value in the constructor. So you may not need to consider `unref(new Cls())`.

maybe this is enough:
```javascript
new A(x)  // new A(_ctx.x)
new A(new B())  // new A(new B())
new A(new B(),x) // new A(new B(),_ctx.x)
```
just modify the code as below:
```javascript
if (needPrefix && !isLocal) {
  //...

  // avoid rewrite class name
  if (
    !parent ||
    parent.type !== 'NewExpression' ||
    (parent.callee as any).name !== id.name
  ) {
    id.name = rewriteIdentifier(id.name, parent, id)
    ids.push(id as QualifiedId)
  }
}
```

Reply:
I think if I don't consider these, I can use my first commit to fix this problem #6483, and the final result should be:

```ts
new A(x) // new (_unref(A))(_ctx.x)
new A(new B()) // new (_unref(A))(new (_unref(B))())
new A(new B(), x) // new (_unref(A))(new (_unref(B))(), _ctx.x)
```

Reply:
let's wait for others' input.

Reply:
Any update on the matter? Just spent whole day debugging this.

Reply:
Any updates?